core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java

```diff
@@ -2615,6 +2616,100 @@ public void tableCreationWithoutNamespace() {
         .hasMessageContaining("Namespace does not exist: non-existing");
   }
 
+  @Test
+  public void testRegisterTable() {
+    C catalog = catalog();
+
+    if (requiresNamespaceCreate()) {
+      catalog.createNamespace(TABLE.namespace());
+    }
+
+    Map<String, String> properties =
+        ImmutableMap.of("user", "someone", "created-at", "2023-01-15T00:00:01");
+    Table table =
+        catalog
+            .buildTable(TABLE, SCHEMA)
+            .withPartitionSpec(SPEC)
+            .withSortOrder(WRITE_ORDER)
+            .withProperties(properties)
+            .create();
+
+    table.newFastAppend().appendFile(FILE_A).commit();
+    table.newFastAppend().appendFile(FILE_B).commit();
+    table.newDelete().deleteFile(FILE_A).commit();
+    table.newFastAppend().appendFile(FILE_C).commit();
+
+    TableOperations ops = ((HasTableOperations) table).operations();
+    String metadataLocation = ops.current().metadataFileLocation();
+
+    catalog.dropTable(TABLE, false);
+
+    Table registeredTable = catalog.registerTable(TABLE, metadataLocation);
+    Assertions.assertThat(registeredTable).isNotNull();
+
+    TableOperations registeredTableOps = ((HasTableOperations) table).operations();
+    Assertions.assertThat(metadataLocation)
+        .isEqualTo(registeredTableOps.current().metadataFileLocation());
```

Comment:
This isn't necessarily true. It's fine for the catalog to import and create its own metadata file copy. I think what we want to validate is that the table is identical, not that it uses the original metadata location.

Reply:
Hi @rdblue, what is your thought here?

I see in https://github.com/apache/iceberg/pull/6742, It was required to `registerTable API to avoid creating new metadata file`.

> remove the commit part in registerTable API to avoid creating new metadata file

Reply:
I don't think that register table should be required to use the same metadata file. Some metadata may be overridden so there's no guarantee it's the same file.